include/fmt/chrono.h

```diff
@@ -332,27 +346,13 @@ auto write(OutputIt out, const std::tm& time, const std::locale& loc,
     using code_unit = char32_t;
 #endif
 
-    using codecvt = std::codecvt<code_unit, char, std::mbstate_t>;
-#if FMT_CLANG_VERSION
-#  pragma clang diagnostic push
-#  pragma clang diagnostic ignored "-Wdeprecated"
-    auto& f = std::use_facet<codecvt>(loc);
-#  pragma clang diagnostic pop
-#else
-    auto& f = std::use_facet<codecvt>(loc);
-#endif
+    constexpr size_t unit_buf_size = 32;
+    code_unit unit_buf[unit_buf_size] = {};
```

Comment:
Let's move buffer management to `[do_]write_codecvt` by making it return a struct, something like:

```c++
struct codecvt_result {
  code_unit buf[32];
  code_unit* end;
};
```

Reply:
Why? ``write_codecvt`` does not depend on the the output buffer size. So I moved the buffer management out of ``write_codecvt``.

Reply:
Because otherwise it has to be done by the callers effectively duplicating the logic.

Reply:
Done

Reply:
``end`` value is dependent of ``buf``.
Returning ``codecvt_result`` from a function corrupts ``end``.